### PR TITLE
Update PHP to allow caller to set curl options

### PR DESCRIPTION
**Pull request by tzs**
Many production environments restrict outgoing network access on their servers that deal with payments, only allowing such access through proxies that vet the destinations before allowing access. This patch provides an option for the caller of verifyIPN() to provide the curl object for verifyIPN() to use instead of allocating its own curl object. This allows the caller to set any proxy options, or other options, that may be required in order to allow the connection back to Paypal to work.

I first tried adding methods to the PaypalIPN class for setting proxy options, but PHP curl has a very large number of proxy settings. The majority of cases in practice could probably be handled with only two or three options, but the approach proposed in this pull request handles all the cases, so I abandoned the other approach. If anyone wants to see that approach, it is on a branch in my clone of this repository, available [here](https://github.com/tzs/ipn-code-samples/tree/tzs-patch-1)

### CODE REVIEW

File: php/PaypalIPN.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ public function getPaypalUri()
      * @return bool
      * @throws Exception
      */
-    public function verifyIPN()
+    public function verifyIPN($ch = null)
```
Comment (overint) [inline php/PaypalIPN.php]: I think this would be better passed in a constructor

File: php/PaypalIPN.php [hunk 2]
```diff
@@ -96,7 +96,9 @@ public function verifyIPN()
         }
 
         // Post the data back to PayPal, using curl. Throw exceptions if errors occur.
-        $ch = curl_init($this->getPaypalUri());
+        if (! isset($ch))
```
Comment (overint) [inline php/PaypalIPN.php]: Missing braces around the if body.

Comment (overint) [reply php/PaypalIPN.php]: Maybe do

```
$ch = isset($this->curlHandler) ? $this->curlHandler : curl_init();
```

**Comment by tzs** [comment]
Good idea on doing it in a constructor. I'll withdraw this pull request and submit another one that takes that approach, essentially exactly as you suggested except it will be curl_handler instead of curlHandler, because the existing code seem to use snake_case for member and local variables rather than camelCase.

**Pull request closed without merging**